datafusion/expr/src/udaf.rs

```diff
@@ -507,6 +510,21 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
     fn coerce_types(&self, _arg_types: &[DataType]) -> Result<Vec<DataType>> {
         not_impl_err!("Function {} does not implement coerce_types", self.name())
     }
+
+    /// Dynamic equality. Allows customizing the equality of aggregate UDFs.
+    /// By default, compares the UDF name and signature.
```

Comment:
Here is a suggestion to improve the docstring

```suggestion
    /// Return true if this aggregate UDF is equal to the other.
    /// 
    /// Allows customizing the equality of aggregate UDFs. Must be consistent
    /// with [`Self::hash_value`].
    ///
    /// By default, compares [`Self::name`] and [`Self::signature`]
```